models/permutation_importance.py

```diff
@@ -15,57 +15,6 @@
 hep.style.use("CMS")
 
 
-def weigted_corrcoef(x, y, w):
-    return 1 - correlation(x, y, w)
-
-def get_corrcoef(vars, weights):
-    w = weights
-    corrcoeff_matrix = np.zeros([vars.shape[1], vars.shape[1]])
-    print(f"INFO: Calculating correlation coefficients for {vars.shape[1]} variables")
-    for i in range(vars.shape[1]):
-        print(f"INFO: Calculating correlation coefficients for variable {i+1}")
-        for j in range(vars.shape[1]):
-            if i <= j:
-                x = vars[:, i]
-                y = vars[:, j]
-
-                # remove event x and y events with valus == -9
-                mask = (x != -9) & (y != -9)
-                #if sum(~mask) > 0:
-                #    print(i, j, sum(~mask))
-            
-                corrcoeff_matrix[i][j] = weigted_corrcoef(x[mask], y[mask], w[mask])
-                corrcoeff_matrix[j][i] = corrcoeff_matrix[i][j]
-
-    return corrcoeff_matrix
-
-def plot_corrcoef(corrcoeff_matrix, var_names, figname, annotate_greater_than=0.2, annotate_less_than=-0.2):
-
-    print("INFO: Plotting correlation matrix")
-    corrcoeff_matrix = corrcoeff_matrix.astype(float)
-
-    fig, ax = plt.subplots(figsize=(50, 50)) 
-    cax = ax.matshow(corrcoeff_matrix, cmap="coolwarm", vmin=-1, vmax=1)
-    colorbar = fig.colorbar(cax)
-    colorbar.set_label("Correlation Coefficient", fontsize=18)
-    ax.set_xticks(range(len(var_names)))
-    ax.set_yticks(range(len(var_names)))
-    ax.set_xticklabels(var_names, fontsize=15, rotation=90)
-    ax.set_yticklabels(var_names, fontsize=15)
-    ax.xaxis.set_ticks_position('bottom')
-    ax.xaxis.set_label_position('bottom')
-
-    #for (i, j), val in np.ndenumerate(corrcoeff_matrix):
-    #    ax.text(j, i, f"{val:.2f}", ha="center", va="center", color="black", fontsize=12)
-    for (i, j), val in np.ndenumerate(corrcoeff_matrix):
-        if (val > annotate_greater_than) or (val < annotate_less_than):
-            ax.text(j, i, f"{val:.2f}", ha="center", va="center", color="black", fontsize=8)
-
-    plt.savefig(f"{figname}.png", dpi=300, bbox_inches="tight")
-    plt.show()
-
-
-
```

Comment:
I guess I understand why you remove the permutation feature importance below but why this as well?